include/seqan3/io/structure/structure_file_in.hpp

```diff
@@ -0,0 +1,1217 @@
+ * that you want to read from and/or if you cannot use file-extension based detection, but know that your input
+ * file has a certain format.
+ *
+ * In most cases the template parameters are deduced completely automatically:
+ * ```cpp
+ * structure_file_in sf{"/tmp/my.dbn"}; // Dot bracket with RNA sequences assumed, regular std::ifstream taken as stream
+ * ```
+ *
+ * Reading from an std::istringstream:
+ * ```cpp
+ * std::string input
+ * {
+ *     "> S.cerevisiae_tRNA-PHE M10740/1-73\n"
+ *     "GCGGAUUUAGCUCAGUUGGGAGAGCGCCAGACUGAAGAUUUGGAGGUCCUGUGUUCGAUCCACAGAAUUCGCA\n"
+ *     "(((((((..((((........)))).((((.........)))).....(((((.......)))))))))))). (-17.50)\n"
+ *     "> example\n"
+ *     "UUGGAGUACACAACCUGUACACUCUUUC\n"
+ *     "..(((((..(((...)))..)))))... (-3.71)\n"
+ * };
+ *
+ * std::istringstream iss(input);
+ *
+ * structure_file_in fin{std::move(iss), structure_file_format_dot_bracket{}};
+ * //              ^ no need to specify the template arguments
+ * ```
+ *
+ * Note that this is not the same as writing `structure_file_in<>` (with angle brackets). In the latter case they are
+ * explicitly set to their default values, in the former case
+ * [automatic deduction](http://en.cppreference.com/w/cpp/language/class_template_argument_deduction) happens which
+ * chooses different parameters depending on the constructor arguments. For opening from file, `structure_file_in<>`
+ * would have also worked, but for opening from stream it would not have.
+ *
+ * In some cases, you do need to specify the arguments, e.g. if you want to read amino acids:
+ *
+ * ```cpp
+ * structure_file_in<structure_file_default_traits_aa> fin{"/tmp/my.dbn"};
+ * ```
+ *
+ * You can define your own traits type to further customise the types used by and returned by this class, see
+ * seqan3::structure_file_default_traits_rna for more details. As mentioned above, specifying at least one
+ * template parameter yourself means that you loose automatic deduction so if you want to read amino acids **and**
+ * want to read from a string stream you need to give all types yourself:
+ *
+ * ```cpp
+ *
+ *  // ... input had amino acid sequences
+ * std::istringstream iss(input);
+ *
+ * structure_file_in<structure_file_default_traits_aa,
+ *                   fields<field::SEQ, field::ID, field::STRUCTURE>,
+ *                   type_list<structure_file_format_dot_bracket>,
+ *                   std::istringstream> fin{std::move(iss), structure_file_format_dot_bracket{}};
+ * ```
+ *
+ * ### Reading record-wise
+ *
+ * You can iterate over this file record-wise:
+ *
+ * ```cpp
+ * structure_file_in fin{"/tmp/my.dbn"};
+ *
+ * for (auto & rec : fin)
+ * {
+ *     std::cout << "ID:  " << get<field::ID>(rec) << '\n';
+ *     std::cout << "SEQ: " << (get<field::SEQ>(rec) | view::to_char) << '\n'; // sequence is converted to char on-the-fly
+ *     std::cout << "STRUCTURE: " << (get<field::STRUCTURE>(rec) | view::to_char) << '\n';
```

Comment:
I think `field::STRUCTURE` is not default anymore, right? So this example wouldn't work. Please double-check also the other examples!

Reply:
The field::STRUCTURE becomes default now, to be consistent with structure_file_out and also to be similar to sequence_io. I think it's more intuitive to use it this way.